packages/terra-button-group/src/ButtonGroup.jsx

```diff
@@ -3,92 +3,50 @@ import PropTypes from 'prop-types';
 import classNames from 'classnames/bind';
 import 'terra-base/lib/baseStyles';
 import ButtonGroupButton from './ButtonGroupButton';
+import ButtonGroupUtils from './ButtonGroupUtils';
 import styles from './ButtonGroup.scss';
 
 const cx = classNames.bind(styles);
 
 const propTypes = {
-  /**
-   * Button group button components that should be grouped together
-   */
-  buttons: PropTypes.arrayOf(PropTypes.element),
-
   /**
    * Child nodes
    */
   children: PropTypes.node,
 
   /**
-   * Indicates if the buttons should have reduced top and bottom padding
-   */
-  isCompact: PropTypes.bool,
-
-  /**
-   * Indicates if the button group should have toggle-style selectability
-   */
-  isSelectable: PropTypes.bool,
-
-  /**
-   * Callback function when the state changes
+   * Callback function when the state changes. Parameters are (event, key).
    */
   onChange: PropTypes.func,
 
   /**
-   * Sets the button size. One of `tiny`, `small`, `medium`, `large`, `huge`
-   */
-  size: PropTypes.oneOf(['tiny', 'small', 'medium', 'large', 'huge']),
-
-  /**
-   * Sets the button group style variation. One of `defaut` or `secondary`
+   * A list of keys of the buttons that should be selected.
    */
-  variant: PropTypes.oneOf(['default', 'secondary']),
+  selectedKeys: PropTypes.arrayOf(PropTypes.string),
```

Comment:
This is probably worth getting a few opinions on. Does anyone have any thoughts on the selectedKeys api vs an explicit selection on the buttons?
```
<ButtonGroup selectedKeys={['one']}>
  <ButtonGroup.Button key="one" />
  <ButtonGroup.Button key="two" />
  <ButtonGroup.Button key="three" />
</ButtonGroup>
```
vs
```
<ButtonGroup>
  <ButtonGroup.Button isSelected />
  <ButtonGroup.Button />
  <ButtonGroup.Button />
</ButtonGroup>
```

Reply:
This first one, using the selectedKeys prop, seems like it would be easier to do multiple selected keys for the multi-select button group from the perspective of the parent component which is rendering the controlled button group.

Reply:
Agree with Brett. The first option is also more consistent with Tabs, List, and Table. The second option is more modular but would requires a relatively more complicated structure on the parent component to map the selectedKey provided in onChange to setting the isSelected prop for each button. For these reasons, I prefer the first option.